data/dataset.py

```diff
@@ -112,8 +121,77 @@ def _build_dataset(self) -> tuple[list[dict], list[dict]]:
 
         return processed_records, unprocessed_records
 
-    def _build_vocab(self) -> tuple[list[str], list[str]]:
-        vocab_src, vocab_tgt = ["<s>", "</s>"], ["<s>", "</s>"]
+    def build_vocab(self) -> tuple[list[str], list[str]]:
+        vocab_src, vocab_tgt = ["<s>", "<blank>", "</s>"], ["<s>", "<blank>", "</s>"]
+        iterators = {
+            "pitch": range(21, 109),
+            "duration_bin": range(self.quantizer.n_duration_bins),
+            "dstart_bin": range(self.quantizer.n_dstart_bins),
+            "velocity_bin": range(self.quantizer.n_velocity_bins),
+        }
+
+        src_product = itertools.product(
+            iterators[self.tokenizer_src.keys[0]],
+            iterators[self.tokenizer_src.keys[1]],
+        )
+
+        tgt_product = itertools.product(
+            iterators[self.tokenizer_tgt.keys[0]],
+            iterators[self.tokenizer_tgt.keys[1]],
+        )
+
+        for val1, val2 in src_product:
+            key = f"{val1}-{val2}"
+            vocab_src.append(key)
+        for val1, val2 in tgt_product:
+            key = f"{val1}-{val2}"
+            vocab_tgt.append(key)
+
+        return vocab_src, vocab_tgt
+
+    def __getitem__(self, idx: int):
+        return self.samples[idx]
+
+    def __len__(self):
+        return len(self.samples)
+
+
+class BinsToVelocityDataset(TokenizedMidiDataset):
+    def __init__(
+        self,
+        split: str = "train",
```

Comment:
```suggestion
        dataset: Dataset,
```

This is important, because after we master the maestro we will start using other MIDI datasets, with exactly the same data structures (compatible with `MidiPiece.from_huggingface`). Please apply this to other dataset classes in this repository - sorry for the trouble, I should've point that out earlier 😶‍🌫️

Reply:
sure!